lang/src/main/kotlin/org/partiql/lang/ast/IsListParenthesizedMeta.kt

```diff
@@ -0,0 +1,20 @@
+package org.partiql.lang.ast
+
+/**
+ * The [IsListParenthesizedMeta] is used to distinguish the two types of [org.partiql.lang.domains.PartiqlAst.Expr.List]
+ * syntax's:
+ *  1. [[ a, b, c ]]
+ *  2. ( a, b, c )
+ *
+ * The existence of this meta on a [org.partiql.lang.domains.PartiqlAst.Expr.List] node indicates that the list uses
+ * the second syntax above (using parenthesis).
+ */
+public class IsListParenthesizedMeta private constructor() : Meta {
+
+    override val tag = TAG
+
+    companion object {
+        const val TAG = "\$is_list_parenthesized"
+        val instance = IsListParenthesizedMeta()
+    }
+}
```

Comment:
I noticed all the metas are this way ... we're writing Kotlin not Java; let's use objects
```
object IsListMeta : Meta {
    override val tag: String = "\$is_list_parenthesized"
}
```

The generated Java is doing exactly what we are doing by hand
```
public final class IsListMeta implements Meta {

   @NotNull
   private static final String tag;

   // Here's your static instance...
   @NotNull
   public static final IsListMeta INSTANCE;

   @NotNull
   public String getTag() {
      return tag;
   }
   
   // Here's your "companion object"...
   static {
      IsListMeta var0 = new IsListMeta();
      INSTANCE = var0;
      tag = "$is_list_parenthesized";
   }

   // Here's your private constructor...
   private IsListMeta() {
   }

  // redacted
}
```